### DISCUSSION TITLE
MA27 Allocating Memory

### DISCUSSION THREAD

**Discussion by SSummits** [IDAES/idaes-pse#891] [Q&A]
I am using ipopt with the ma27 linear solver for my optimization problem. I am using a pretty big model and so I am frequently running into memory allocation issues that are slowing down the solver (warning message shown below). Is there any way to pre-allocate more memory to ma27 prior to running the solver? The coin-or documentation shows the options to increase the integer and real workspace memory for ma27 in ipopt, but those options appear to be missing in this version of the solver.

MA27BD returned iflag=-4 and requires more memory.
 Increase liw from 13060495 to 26120990 and la from 12405245 to 24820570 and factorize again.

**Comment by adowling2** [comment]
My recommendation is to try ma57. Also, needing to relocate memory can be a symptom of poor modeling scaling.

**Comment by andrewlee94** [comment]
As a general comment, MA27 is generally very good at estimating the amount of memory it will require, and if you are seeing this error frequently, it often indicates an underlying issue with the model itself rather than the solver. Are you seeing a lot of steps in with regularization (values in the ||lg r|| column) and a lot of line search failures?
